### PR TITLE
Try to fix lr hydrostatic balance func for gpus

### PR DESCRIPTION
**Pull request by charleskawczynski**
I think this might be all we need-- it won't be efficient, but I don't think it needs to be since it happens at initialization

### CODE REVIEW

File: src/utils/discrete_hydrostatic_balance.jl [hunk 1]
```diff
@@ -80,8 +80,15 @@ function set_discrete_hydrostatic_balanced_pressure!(
     @. ᶠgradᵥ_ᶜp[colidx] = -(ᶠgradᵥ_ᶜΦ[colidx] * ᶠinterp(ᶜρ[colidx]))
     ᶜp_data = Fields.field_values(ᶜp[colidx])
     ᶠgradᵥ_ᶜp_data = Fields.field_values(ᶠgradᵥ_ᶜp[colidx])
-    ᶜp_data[1] = p1
+    ᶜp_data_lev₋₁ = Spaces.level(p1, 1-1)
     for i in 2:Spaces.nlevels(axes(ᶜp))
-        ᶜp_data[i] = ᶜp_data[i - 1] + ᶠgradᵥ_ᶜp_data[i].u₃
+        ᶜp_data_lev₋₁ = Spaces.level(p1, i-1)
```
Comment (szy21) [inline src/utils/discrete_hydrostatic_balance.jl:85]:
```suggestion
        ᶜp_data_lev₋₁ = Spaces.level(ᶜp_data, i-1)
```

**Review by szy21 — APPROVED** [review]
Thanks! Here are some suggestions based on my understanding. Is this what you meant?

**Comment by charleskawczynski** [comment]
> Is this what you meant?

Yes, thanks for the suggestions. I think that branch was somewhat unfinished. I think this should work (although it won't be very performant). It's only done during initialization, so I think it should be fine. We could also split the CPU/GPU implementations so that the CPU doesn't suffer from this awkward pattern.

**Comment by szy21** [comment]
I think this is fine. As you mentioned it's only used during the initialization. Also it's only used in this one job.

**Review by trontrytel — APPROVED** [review]

**Pull request merged**